safehttp/plugins/fetch_metadata/fetch_metadata_test.go

```diff
@@ -0,0 +1,399 @@
+		"Content-Type":           {"text/plain; charset=utf-8"},
+		"X-Content-Type-Options": {"nosniff"},
+	}
+	if diff := cmp.Diff(wantHeaders, map[string][]string(rec.Header())); diff != "" {
+		t.Errorf("rec.Header() mismatch (-want +got):\n%s", diff)
+	}
+	if want, got := "Forbidden\n", rec.Body(); got != want {
+		t.Errorf("response body got: %q want: %q", got, want)
+	}
+
+	rec = safehttptest.NewResponseRecorder()
+	p.SetReportOnly(logger)
+
+	p.Before(rec.ResponseWriter, req)
+
+	if want, got := safehttp.StatusOK, safehttp.StatusCode(rec.Status()); want != got {
+		t.Errorf("status code got: %v want: %v", got, want)
+	}
+	if diff := cmp.Diff(map[string][]string{}, map[string][]string(rec.Header())); diff != "" {
+		t.Errorf("rec.Header() mismatch (-want +got):\n%s", diff)
+	}
+	if want, got := "", rec.Body(); got != want {
+		t.Errorf("response body got: %q want: %q", got, want)
+	}
+	if want := "POST /pizza"; logger.report != want {
+		t.Errorf("logger.report: got %s, want %s", logger.report, want)
+	}
+}
+
+func TestEnableDisableNavIsolation(t *testing.T) {
+	req := safehttptest.NewRequest("GET", "https://spaghetti.com/pizza", nil)
+	req.Header.Add("Sec-Fetch-Site", "cross-site")
+	req.Header.Add("Sec-Fetch-Mode", "navigate")
+	req.Header.Add("Sec-Fetch-Dest", "image")
+	rec := safehttptest.NewResponseRecorder()
+
+	p := fetchmetadata.NewPlugin()
+	p.NavIsolation = true
+	p.Before(rec.ResponseWriter, req)
+
+	if want, got := safehttp.StatusForbidden, safehttp.StatusCode(rec.Status()); want != got {
+		t.Errorf("status code got: %v want: %v", got, want)
+	}
+	wantHeaders := map[string][]string{
+		"Content-Type":           {"text/plain; charset=utf-8"},
+		"X-Content-Type-Options": {"nosniff"},
+	}
+	if diff := cmp.Diff(wantHeaders, map[string][]string(rec.Header())); diff != "" {
+		t.Errorf("rec.Header() mismatch (-want +got):\n%s", diff)
+	}
+	if want, got := "Forbidden\n", rec.Body(); got != want {
+		t.Errorf("response body got: %q want: %q", got, want)
+	}
+
+	rec = safehttptest.NewResponseRecorder()
+
+	p.NavIsolation = false
+	p.Before(rec.ResponseWriter, req)
+
+	if want, got := safehttp.StatusOK, safehttp.StatusCode(rec.Status()); want != got {
+		t.Errorf("status code got: %v want: %v", got, want)
+	}
+	if diff := cmp.Diff(map[string][]string{}, map[string][]string(rec.Header())); diff != "" {
+		t.Errorf("rec.Header() mismatch (-want +got):\n%s", diff)
+	}
+	if want, got := "", rec.Body(); got != want {
+		t.Errorf("response body got: %q want: %q", got, want)
+	}
+}
+
+func TestCORSEndpoint(t *testing.T) {
+	methods := []string{"GET", "POST"}
+	for _, m := range methods {
+		req := safehttptest.NewRequest(m, "https://spaghetti.com/pizza", nil)
+		req.Header.Add("Sec-Fetch-Site", "cross-site")
+		req.Header.Add("Sec-Fetch-Mode", "navigate")
+		req.Header.Add("Sec-Fetch-Dest", "image")
+		rec := safehttptest.NewResponseRecorder()
+		p := fetchmetadata.NewPlugin("https://spaghetti.com/pizza")
+		p.NavIsolation = true
```

Comment:
I think this line shouldn't be here probably.

Reply:
It should be because one of the requests breaks the navigation isolation policy and the other one the resource isolation policy. I configured the plugin in the most strict way.

Reply:
Ah, I see, nice! :)